File: src/1.Arrays_and_hashing/3.Two_sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> dict = new Dictionary<int, int>();
        for(int i = 0; i < nums.Length; i++)
        {
          int num = nums[i];
          int diff = target - num;
          bool hasDiff = dict.ContainsKey(diff);
          if(hasDiff)
          {
            return new int[]{dict[diff], i};
          }
          if(!dict.ContainsKey(num))
          {
            dict.Add(num, i);
          }
        }
        return new int[]{0,1};
    }
}

File: src/1.Arrays_and_hashing/3.Two_sum.py
# https://leetcode.com/problems/two-sum/

def twoSum(self, nums, target):
    dict = {}
    for i in range(len(nums)):
        num = nums[i]
        diff = target - num
        hasDiff = diff in dict
        if hasDiff:
            return [dict[diff], i]
        if num not in dict:
            dict[num] = i
    return [0,1]